src/common/hist_util.cc

```diff
@@ -830,54 +830,78 @@ void GHistIndexBlockMatrix::Init(const GHistIndexMatrix& gmat,
 /*!
  * \brief fill a histogram by zeros in range [begin, end)
  */
-void InitilizeHistByZeroes(GHistRow hist, size_t begin, size_t end) {
+template<typename GradientSumT>
+void InitilizeHistByZeroes(GHistRow<GradientSumT> hist, size_t begin, size_t end) {
 #if defined(XGBOOST_STRICT_R_MODE) && XGBOOST_STRICT_R_MODE == 1
-  std::fill(hist.begin() + begin, hist.begin() + end, tree::GradStats());
+  std::fill(hist.begin() + begin, hist.begin() + end,
+            xgboost::detail::GradientPairInternal<GradientSumT>());
 #else  // defined(XGBOOST_STRICT_R_MODE) && XGBOOST_STRICT_R_MODE == 1
-  memset(hist.data() + begin, '\0', (end-begin)*sizeof(tree::GradStats));
```

Comment:
Can we use `std::fill` consistently?

Reply:
we have to consider it separately as performance can be affected.
https://github.com/dmlc/xgboost/issues/5579